src/inet/TCPEndPointImplSockets.cpp

```diff
@@ -69,9 +69,12 @@ namespace Inet {
 
 CHIP_ERROR TCPEndPointImplSockets::BindImpl(IPAddressType addrType, const IPAddress & addr, uint16_t port, bool reuseAddr)
 {
-    CHIP_ERROR res = GetSocket(addrType);
+    ReturnErrorOnFailure(GetSocket(addrType));
+
+    // need to be in a connected state to be able to bind
+    VerifyOrReturnError(mSocket >= 0, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
But here's the thing: GetSocket already enforces that invariant, as it should.

Reply:
Yes, the code was always correct. The reason I have this is that clang-tidy cannot check for these guarantees. It looks like this:

![image](https://github.com/user-attachments/assets/11fdacff-eb36-45d2-bda1-c31774c5ae18)

So I can either tell clang-tidy to ignore (I can add nolint entries throughout) or I can add extra and redundant checks. Since this is a `sockets` implementation, it should be for a posix-like system so I was less worried about flash space, hence extra redundant checks seemed ok.

Reply:
And the explanation is clearly wrong: we return CHIP_ERROR_POSIX and then tidy says "what if we have IsSuccess returns true on it" which are not compatible ... but tidy cannot know because IsSuccess is some external function that in theory could change.

So the fact that this is correct is our internal knowledge. Either I do checks or NOLINT.

Reply:
I would suggest NOLINT with comment and a followup issue to restructure this code to make the invariants clearer?

Reply:
I would suggest NOLINT as well here. But then, if clang-tidy cannot interpret ReturnErrorOnFailure() correctly, this should be a more widespread problem with it, isn't it?

Reply:
Ok, switched to NOLINT.

Regarding clang-tidy, I believe it will only be when clang can detect specific code usages that it thinks it is wrong. The tool still has limited static analyzer checking, so our code base does not fully blow up.